File: src/arduinobot_test/scripts/simple_service.py
#!/usr/bin/python3
import rospy
from arduinobot_test.srv import AddTwoInts, AddTwoIntsResponse

def add_two_ints_cb(req):
    rospy.loginfo("ready to sum %d and %d", req.a, req.b)
    return AddTwoIntsResponse(req.a + req.b)

if __name__ == '__main__':
    rospy.init_node("simpler_service")
    service = rospy.Service('add_two_ints', AddTwoInts, add_two_ints_cb)
    rospy.loginfo("The service is ready to add ints")
    rospy.spin()

File: src/arduinobot_test/src/simple_service.cpp
# include <ros/ros.h>
# include <arduinobot_test/AddTwoInts.h>


bool add_two_ints(arduinobot_test::AddTwoInts::Request &req,
                  arduinobot_test::AddTwoInts::Response &res)
{
    ROS_INFO("ready to sum %d and %d", req.a, req.b);
    res.sum = req.a + req.b;
    return true;
}

int main(int argc, char **argv){
    ros::init(argc, argv, "simple_service");
    ros::NodeHandle n;
    ros::ServiceServer service = n.advertiseService("add_two_ints", add_two_ints);
    ROS_INFO("The service is ready to add ints");
    ros::spin();
    return 0;
}